lib/domain/models/resource.dart

```diff
@@ -14,6 +14,11 @@ class Resource extends KeyStorable {
   /// This is useful in cases where we must match a `Deck.tags` with each available resources
   final List<String> tags;
 
+  /// Metadata that describes which type of [url] this resource refers to
+  ///
+  /// Example types: "video", "article", etctera.
+  final String type;
```

Comment:
Why using a String instead a enum? I mean, we will have to customize this Resource in the UI, so we will need to compare these types

Reply:
No need to force an enumerator here. The rationale is that we can have a default fallback case if none is match (like an default icon/ui customization). If you require an enumerator here, we may have to crash at runtime when reading such resources (or even make this type optional). So, in my opinion it's better to iterate over a loose type instead of the enumerator in this particular case.

Reply:
@matuella  We can merge both solutions: an enum with a "unknown" type. This way we doesn't need to crash at runtime nor compare strings constants to create a custom UI. My points against using a `String` here are 1. We will have silent type inconsistencies (we will show a default UI behavior, but someone see's it, we will neve know that a Resource is with the wrong type) and 2. We will have to compare string constants somewhere to customize the UI, like `if (type == "book") { ... }`

Reply:
I think this is more of a personal opinion than what we should seek here, but I guess the work is the same with enumerators and strings